### PR TITLE
Added cpuid_get for STM boards

### PR DESCRIPTION
**Pull request by jhollister**
Adds cpuid_get for the rest of the STM boards as per issue #1511. All of the values were obtained from reference manuals for the corresponding board but I wasn't able to test each function.

### CODE REVIEW

File: cpu/stm32f4/periph/cpuid.c [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2014 FU Berlin
```
Comment (LudwigOrtmann) [inline cpu/stm32f4/periph/cpuid.c]: I guess you overlooked to change this?

File: cpu/stm32f3/periph/cpuid.c [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2014 FU Berlin
```
Comment (LudwigOrtmann) [inline cpu/stm32f3/periph/cpuid.c]: Compare https://github.com/RIOT-OS/RIOT/pull/2552/files#r25995093

File: cpu/stm32f0/periph/cpuid.c [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2014 FU Berlin
```
Comment (LudwigOrtmann) [inline cpu/stm32f0/periph/cpuid.c]: Compare https://github.com/RIOT-OS/RIOT/pull/2552/files#r25995093

File: cpu/stm32f4/periph/cpuid.c (hunk 1 above)
Comment (jhollister) [inline cpu/stm32f4/periph/cpuid.c]: I did. Does the date only need to be updated or should the location be changed also?

Comment (jhollister) [inline cpu/stm32f4/periph/cpuid.c]: Also, for changes like this do you prefer to keep it in a single commit or make a new commit?

File: cpu/stm32f4/periph/cpuid.c [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+#include <string.h>
+#include "cpu-conf.h"
+
+#include "periph/cpuid.h"
+
+void cpuid_get(void *id)
+{
+    memcpy(id, (void *)(0x1fff7A10), CPUID_ID_LEN);
```
Comment (gebart) [inline cpu/stm32f4/periph/cpuid.c]: Use a symbolic name for the source register, preferably one from the CPU header file if it is defined there.

File: cpu/stm32f4/periph/cpuid.c (hunk 1 above)
Comment (thomaseichinger) [inline cpu/stm32f4/periph/cpuid.c]: Best practice is to use one commit per module i.e. CPU in this case and add commits with review changes prefixed by `SQUASHME` or `FIX`. These have be squashed when ACKed before merging.

**Comment by jhollister** [comment]
I couldn't find a symbolic name in the CPU header files. Is it ok to add one where I did or would it be better in `cpuconf.h`?

**Comment by thomaseichinger** [comment]
@jhollister Using local defines is fine, please squash. Mind to split up the commit to three separate with message prefix `cpu/stm32fn: ...` with n = 0,3,4?

**Comment by jhollister** [comment]
@thomaseichinger Done.

File: cpu/stm32f0/periph/cpuid.c [hunk 5]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2015 FU Berlin
```
Comment (LudwigOrtmann) [inline cpu/stm32f0/periph/cpuid.c]: Unless you are employed at FU Berlin, please grab the copyright.

File: cpu/stm32f0/periph/cpuid.c [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @addtogroup  driver_periph
+ * @{
+ *
+ * @file        cpuid.c
```
Comment (LudwigOrtmann) [inline cpu/stm32f0/periph/cpuid.c]: Remove the filename please (just `@file`).

**Comment by jhollister** [comment]
@LudwigOrtmann Copyright and filename are fixed.

**Comment by fnack** [comment]
I kicked Travis.

@thomaseichinger: Is there something missing for this PR. Otherwise it would be great, if you could ACK and merge soon. Then I could include the cpuid feature in #2578.

**Comment by LudwigOrtmann** [comment]
I guess this also works for the `nucleo-f334`?

**Comment by LudwigOrtmann** [comment]
ACK from my side.

The nucleo support can come as a follow up PR as well .. I'm not sure how to verify the validity of the CPUID, but at least it prints out something.

**Pull request merged**